homeassistant/components/unifi/image.py

```diff
@@ -0,0 +1,136 @@
+"""Image platform for UniFi Network integration.
+
+Support for QR code for guest WLANs.
+"""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Generic
+
+import aiounifi
+from aiounifi.interfaces.api_handlers import ItemEvent
+from aiounifi.interfaces.wlans import Wlans
+from aiounifi.models.api import ApiItemT
+from aiounifi.models.wlan import Wlan
+
+from homeassistant.components.image import DOMAIN, ImageEntity, ImageEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceEntryType
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+import homeassistant.util.dt as dt_util
+
+from .const import ATTR_MANUFACTURER, DOMAIN as UNIFI_DOMAIN
+from .controller import UniFiController
+from .entity import HandlerT, UnifiEntity, UnifiEntityDescription
+
+
+@callback
+def async_wlan_qr_code_image_fn(controller: UniFiController, wlan: Wlan) -> bytes:
+    """Calculate receiving data transfer value."""
+    return controller.api.wlans.generate_wlan_qr_code(wlan)
+
+
+@callback
+def async_wlan_device_info_fn(api: aiounifi.Controller, obj_id: str) -> DeviceInfo:
+    """Create device registry entry for WLAN."""
+    wlan = api.wlans[obj_id]
+    return DeviceInfo(
+        entry_type=DeviceEntryType.SERVICE,
+        identifiers={(DOMAIN, wlan.id)},
+        manufacturer=ATTR_MANUFACTURER,
+        model="UniFi Network",
+        name=wlan.name,
+    )
+
+
+@dataclass
+class UnifiImageEntityDescriptionMixin(Generic[HandlerT, ApiItemT]):
+    """Validate and load entities from different UniFi handlers."""
+
+    image_fn: Callable[[UniFiController, ApiItemT], bytes]
+    value_fn: Callable[[ApiItemT], str]
+
+
+@dataclass
+class UnifiImageEntityDescription(
+    ImageEntityDescription,
+    UnifiEntityDescription[HandlerT, ApiItemT],
+    UnifiImageEntityDescriptionMixin[HandlerT, ApiItemT],
+):
+    """Class describing UniFi image entity."""
+
+
+ENTITY_DESCRIPTIONS: tuple[UnifiImageEntityDescription, ...] = (
+    UnifiImageEntityDescription[Wlans, Wlan](
+        key="WLAN QR Code",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        has_entity_name=True,
+        entity_registry_enabled_default=False,
```

Comment:
It is already marked as "diagnostic", any reason we disable this by default?

Reply:
Only that it provides access to all networks without any additional decision. You think we should skip disabling it?

Reply:
Removed disabled by default

Reply:
After additional discussions on discord we agreed to keep it as is.